build.rs

```diff
@@ -28,6 +28,22 @@ fn main() {
         println!("cargo:rustc-cfg=feature=\"mem\"");
     }
 
+    let target_os = env::var("CARGO_CFG_TARGET_OS").unwrap();
+    if target_os != "windows" {
+        let target_arch = env::var("CARGO_CFG_TARGET_ARCH").unwrap();
+        if target_arch == "x86_64" {
+            let cfg = &mut cc::Build::new();
```

Comment:
This compilation will want to be in the `c` module as well, and if possible it'd be great to use the same compilation support that all the other C intrinsics use below.

Reply:
Indeed, it makes sense to refactor that way.